src/routes/accounts/counterfactual-safes/counterfactual-safes.controller.spec.ts

```diff
@@ -400,6 +400,261 @@ describe('CounterfactualSafesController', () => {
+          .build(),
+      ]);
+      counterfactualSafesDataSource.getCounterfactualSafesForAccount.mockResolvedValue(
+        counterfactualSafes,
+      );
+
+      await request(app.getHttpServer())
+        .get(`/v1/accounts/${address}/counterfactual-safes`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(200)
+        .expect(
+          counterfactualSafes.map((counterfactualSafe) => ({
+            chainId: counterfactualSafe.chain_id,
+            creator: counterfactualSafe.creator,
+            fallbackHandler: counterfactualSafe.fallback_handler,
+            owners: counterfactualSafe.owners,
+            predictedAddress: counterfactualSafe.predicted_address,
+            saltNonce: counterfactualSafe.salt_nonce,
+            singletonAddress: counterfactualSafe.singleton_address,
+            threshold: counterfactualSafe.threshold,
+          })),
+        );
+
+      expect(
+        counterfactualSafesDataSource.getCounterfactualSafesForAccount,
+      ).toHaveBeenCalledWith(account);
+      expect(
+        counterfactualSafesDataSource.getCounterfactualSafesForAccount,
+      ).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not propagate a database error', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const chain = chainBuilder().build();
+      const authPayloadDto = authPayloadDtoBuilder()
+        .with('chain_id', chain.chainId)
+        .with('signer_address', address)
+        .build();
+      const accessToken = jwtService.sign(authPayloadDto);
+      const account = accountBuilder().build();
+      const accountDataTypes = [
+        accountDataTypeBuilder()
+          .with('name', AccountDataTypeNames.CounterfactualSafes)
+          .with('is_active', true)
+          .build(),
+      ];
+      accountsRepository.getAccount.mockResolvedValue(account);
+      accountsRepository.getDataTypes.mockResolvedValue(accountDataTypes);
+      accountsRepository.getAccountDataSettings.mockResolvedValue([
+        accountDataSettingBuilder()
+          .with('account_id', account.id)
+          .with('account_data_type_id', accountDataTypes[0].id)
+          .with('enabled', true)
+          .build(),
+      ]);
+      counterfactualSafesDataSource.getCounterfactualSafesForAccount.mockRejectedValue(
+        new Error('Database error.'),
+      );
+
+      await request(app.getHttpServer())
+        .get(`/v1/accounts/${address}/counterfactual-safes`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(500)
+        .expect({
+          code: 500,
+          message: 'Internal server error',
+        });
+
+      expect(
+        counterfactualSafesDataSource.getCounterfactualSafesForAccount,
+      ).toHaveBeenCalledWith(account);
+      expect(
+        counterfactualSafesDataSource.getCounterfactualSafesForAccount,
+      ).toHaveBeenCalledTimes(1);
+    });
+
+    it('returns 403 if no token is present', async () => {
```

Comment:
We need to test the authorisation, e.g. if the token doesn't match the address in the URL.